pennylane/vqe/vqe.py

```diff
@@ -73,54 +78,65 @@ class Hamiltonian:
+    >>> qml.Hamiltonian([1.], [qml.PauliX(0)]) + 2 * qml.PauliZ(0) @ qml.PauliZ(1)
 
-        >>> qml.PauliX(0) + 2 * qml.PauliZ(0) @ qml.PauliZ(1)
+    is equivalent to the following Hamiltonian:
 
-        is equivalent to the following Hamiltonian:
+    >>> qml.Hamiltonian([1, 2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1)])
 
-        >>> qml.Hamiltonian([1, 2], [qml.PauliX(0), qml.PauliZ(0) @ qml.PauliZ(1)])
+    While scalar multiplication requires native python floats or integer types,
+    addition, subtraction, and tensor multiplication of Hamiltonians with Hamiltonians or
+    other observables is possible with tensor-valued coefficients, i.e.:
 
-        When Hamiltonians are defined using arithmetic operations **inside of QNodes**, constituent observables
-        may be queued as operations/an error may be thrown. Thus, Hamiltonians must be defined either outside of QNodes,
-        or inside of QNodes using the conventional method.
+    >>> H1 = qml.Hamiltonian(torch.tensor([1.]), [qml.PauliX(0)])
+    >>> H2 = qml.Hamiltonian(torch.tensor([2., 3.]), [qml.PauliY(0), qml.PauliX(1)])
+    >>> H3 = qml.Hamiltonian(torch.tensor([1., 2., 3.]), [qml.PauliX(0), qml.PauliY(0), qml.PauliX(1)])
+    >>> H3.compare(H1 + H2)
+    True
+
+    .. Warning::
+        When Hamiltonians are defined using arithmetic operations **inside of quantum functions**, constituent observables
+        may be queued as operations/an error may be thrown. Thus, Hamiltonians must be defined either outside
+        of quantum functions, or inside of quantum functions using the conventional construction method.
 
-        Note that this issue also arises when calling the ``simplify()`` method.
     """
-    # Todo: this is a temporary solution to make the circuit drawer work
-    num_params = 0
 
-    def __init__(self, coeffs, observables, simplify=False):
+    num_wires = qml.operation.AnyWires
+    num_params = 1
+    par_domain = "A"
+    grad_method = "A"  # supports analytic gradients
+
+    def __init__(self, coeffs, observables, simplify=False, id=None, do_queue=True):
 
-        if len(coeffs) != len(observables):
+        if qml.math.shape(coeffs)[0] != len(observables):
             raise ValueError(
                 "Could not create valid Hamiltonian; "
                 "number of coefficients and operators does not match."
             )
 
-        if any(np.imag(coeffs) != 0):
-            raise ValueError(
-                "Could not create valid Hamiltonian; " "coefficients are not real-valued."
-            )
-
         for obs in observables:
             if not isinstance(obs, Observable):
                 raise ValueError(
                     "Could not create circuits. Some or all observables are not valid."
                 )
 
-        self._coeffs = list(coeffs)
+        self._coeffs = coeffs
         self._ops = list(observables)
+        self._wires = qml.wires.Wires.all_wires([op.wires for op in self.ops], sort=True)
```

Comment:
nice that the wires class makes this so easy to do!

Reply:
It is rare to hear praise for the wires class ;)

Reply:
I've actually found so many useful things in there while implementing compilation, `shared_wires`, `unique_wires`, etc. :heart: